core/test-runtime/src/lib.rs

```diff
@@ -149,15 +152,57 @@ pub fn changes_trie_config() -> primitives::ChangesTrieConfiguration {
 	}
 }
 
-pub mod api {
-	use system;
-	impl_stubs!(
-		version => |()| super::version(),
-		authorities => |()| system::authorities(),
-		initialise_block => |header| system::initialise_block(header),
-		execute_block => |block| system::execute_block(block),
-		apply_extrinsic => |utx| system::execute_transaction(utx),
-		finalise_block => |()| system::finalise_block(),
-		balance_of => |a| system::balance_of(a)
-	);
+mod test_api {
+	decl_apis! {
+		pub trait TestAPI {
+			fn balance_of<AccountId>(id: AccountId) -> u64;
+		}
+	}
+}
+use test_api::runtime::TestAPI;
+
+struct Runtime;
+
+impl_apis! {
+	impl Core<Block, AuthorityId> for Runtime {
+		fn version() -> RuntimeVersion {
+			version()
+		}
+
+		fn authorities() -> Vec<AuthorityId> {
+			system::authorities()
+		}
+
+		fn execute_block(block: Block) {
+			system::execute_block(block)
+		}
+	}
+
+	impl BlockBuilder<Block, u32, u32> for Runtime {
+		fn initialise_block(header: <Block as BlockT>::Header) {
+			system::initialise_block(header)
+		}
+
+		fn apply_extrinsic(extrinsic: <Block as BlockT>::Extrinsic) -> ApplyResult {
+			system::execute_transaction(extrinsic)
+		}
+
+		fn finalise_block() -> <Block as BlockT>::Header {
+			system::finalise_block()
+		}
+
+		fn inherent_extrinsics(_data: u32) -> Vec<u32> {
```

Comment:
`unimplemented` is a bit concerning. is it ever called? if not, are the others? if not, then maybe split the trait up into two. if so then we probably want to implement in a null fashion (e.g. `return Default::default();`).

Reply:
Hmm, as I understand it is just the test runtime, that is only used for testing. Before that, these functions where not provided a s thus not called. I think for testing purposes it is okay to have `unimplemented!`.

Reply:
i would put in legit impls anyway to make it more correct, but it's not something that needs to happen with this PR as it's essentially unrelated.